elastic/kibana
# [Fleet] index patterns to handle uploaded packages

Resolves https://github.com/elastic/kibana/issues/83312, https://github.com/elastic/kibana/issues/83311

- updates index pattern logic to:
  - check the install source and not reach out to the registry by default (this will be updated to package storage once available)
  - calculate fields for uploaded packages
- removes install source from the cache key.  this may be temporary or we may want to store this as part of package info if end up needing the information later, if we decide to reinstall from cache, we may need it for the saved object property.  once package storage is available, though, i'm not sure it matters what the install_source is.
- updates the uploaded package fixtures (apache_0.1.4.tar.gz and apache_0.1.4.zip) to have a unique field to test against to make sure index patterns from an uploaded package (and not the registry) were installed
- you can now upload a custom package or a package with a version that does not exist in the registry

Test:

- Upload the apache_0.1.4.tar.gz package fixture.  Navigate to index patterns and see they are there
- Upload a custom package that doesn't exist in the registry like `custom_0.1.4.tar.gz` or a version of a package that does not exist in the registry such as `apache_3.1.4.tar.gz`.  There should be no error.  Check the index patterns and search for the fields
